From b630fa6001eb141163de3715f9feb1a4055ab20f Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 30 Oct 2024 13:40:12 +0100
Subject: [PATCH 1459/1695] drivers: clk: stm32mp15: remove unused
 stm32mp1_clk_mcuss_protect()

Remove STM32MP15 platform function stm32mp1_clk_mcuss_protect()
that duplicates existing stm32_rcc_set_mckprot() platforms function.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1e973214ff859218ff767f9c130fbaef9af041b9 drivers: clk: stm32mp15: remove unused stm32mp1_clk_mcuss_protect()]
Change-Id: I2c440ff3202a9490e74365437111e3ae62224037
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/415235
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/clk/clk-stm32mp15.c         | 10 ----------
 core/include/drivers/stm32mp1_rcc_util.h |  3 ---
 2 files changed, 13 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp15.c b/core/drivers/clk/clk-stm32mp15.c
index c2b167793..0b4c54fc4 100644
--- a/core/drivers/clk/clk-stm32mp15.c
+++ b/core/drivers/clk/clk-stm32mp15.c
@@ -1405,16 +1405,6 @@ static void __maybe_unused disable_rcc_tzen(void)
 	io_clrbits32(stm32_rcc_base() + RCC_TZCR, RCC_TZCR_TZEN);
 }
 
-void stm32mp1_clk_mcuss_protect(bool enable)
-{
-	uintptr_t rcc_base = stm32_rcc_base();
-
-	if (enable)
-		io_setbits32(rcc_base + RCC_TZCR, RCC_TZCR_MCKPROT);
-	else
-		io_clrbits32(rcc_base + RCC_TZCR, RCC_TZCR_MCKPROT);
-}
-
 #ifdef CFG_STM32_CPU_OPP
 static bool stm32mp1_clk_pll1_settings_are_valid(void)
 {
diff --git a/core/include/drivers/stm32mp1_rcc_util.h b/core/include/drivers/stm32mp1_rcc_util.h
index 18fa8d3ed..808f47606 100644
--- a/core/include/drivers/stm32mp1_rcc_util.h
+++ b/core/include/drivers/stm32mp1_rcc_util.h
@@ -34,9 +34,6 @@ void stm32mp_register_clock_parents_secure(unsigned long clock_id __unused)
 }
 #endif
 
-/* Protect the MCU clock subsytem */
-void stm32mp1_clk_mcuss_protect(bool enable);
-
 /* Save PLL1 configuration data for low power sequence if any */
 void stm32mp1_clk_lp_save_opp_pll1_settings(uint8_t *data, size_t size);
 
-- 
2.47.2

